### DISCUSSION TITLE
entry.onPublish hook

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#9394] [Ideas] [answered]
> Created by: Florian (fw@demodern.de) on 2015/03/13 06:38:29 +0000
> Votes at time of UserVoice import: 4

-----------------------------------

A hook for when an entry is actually published. onSaveEntry fires when a user saves the entry, but if an entry is set to publish in the future, no event is fired.

This would also be a unique call which is fired only when a user manually publishes a post.

**Comment by brandonkelly** [comment]
[This module](https://github.com/putyourlightson/craft-element-status-events) can be used to track the moment an entry goes live.

**Comment by brandonkelly** [comment, marked as answer]
Just [PR’d a native solution for this](https://github.com/craftcms/cms/pull/17024) for Craft 5.7: there’s a new `staticStatuses` config setting, which can be used to opt into having entry statuses stored statically and only updated on entry save; plus an `update-statuses` CLI command that can be hit with a Cron job, which updates any entries with out-of-date statuses, triggering `save` events and clearing caches in the process. (Plus a new `Entry::$oldStatus` property that can be checked from `save` event handlers to see if the status just changed.)

**Reply by brandonkelly** [reply]
Craft 5.7.0 is out with that change.
